packages/react-native-otlp/ios/RNEmbraceOTLP/RNEmbraceOTLP.swift

```diff
@@ -0,0 +1,223 @@
+      let spanConfig = CustomExporterConfig.fromDictionary(spanConfigDict!)
+      customSpanExporter = self.setOtlpHttpTraceExporter(endpoint: spanConfig.endpoint,
+                                                         timeout: spanConfig.timeout!,
+                                                         header: spanConfig.header!);
+    }
+    
+    // OTLP HTTP Log Exporter
+    if logConfigDict != nil {
+      let logConfig = CustomExporterConfig.fromDictionary(logConfigDict!)
+      customLogExporter = self.setOtlpHttpLogExporter(endpoint: logConfig.endpoint,
+                                                      timeout: logConfig.timeout!,
+                                                      header: logConfig.header!)
+    }
+            
+      return OpenTelemetryExport(spanExporter: customSpanExporter, logExporter: customLogExporter);
+    }
+    
+    @objc(startNativeEmbraceSDK:otlpExportConfigDict:resolver:rejecter:)
+    func startNativeEmbraceSDK(sdkConfigDict: NSDictionary, otlpExportConfigDict: NSDictionary, resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
+        let config = SDKConfig(from: sdkConfigDict)
+
+        DispatchQueue.main.async {
+            do {
+                var embraceOptions: Embrace.Options {
+                    var crashReporter: CrashReporter?
+                    if config.disableCrashReporter {
+                        crashReporter = nil
+                    } else {
+                        crashReporter = EmbraceCrashReporter()
+                    }
+
+                    let servicesBuilder = CaptureServiceBuilder().addDefaults()
+                    if config.disableAutomaticViewCapture {
+                            servicesBuilder.remove(ofType: ViewCaptureService.self)
+                    }
+
+                    var endpoints: Embrace.Endpoints?
+                    if config.endpointBaseUrl != nil {
+                        endpoints = Embrace.Endpoints(baseURL: config.endpointBaseUrl!,
+                                                      developmentBaseURL: config.endpointBaseUrl!,
+                                                      configBaseURL: config.endpointBaseUrl!)
+                    }
+                    
+                    var customExporters: OpenTelemetryExport = self.setHttpExporters(otlpExportConfigDict["traceExporter"] as? NSDictionary,
+                                                                                     logConfigDict: otlpExportConfigDict["logExporter"] as? NSDictionary)
+
+                    return .init(
+                        appId: config.appId,
+                        appGroupId: config.appGroupId,
+                        platform: .reactNative,
+                        endpoints: endpoints,
+                        captureServices: servicesBuilder.build(),
+                        crashReporter: crashReporter,
+                        export: customExporters
```

Comment:
overall I think this approach makes sense, I think the 2 main things to see about would be: 1) is there any way to avoid some of the duplication with the code in `core`and 2) making sure we include code samples for customers to do the custom export themselves if they have their SDK startup defined in native code rather than going through the JS wrapper, similar to what we have for: https://embrace.io/docs/react-native/integration/track-navigation-screens/#disable-auto-tracking-for-native-screens

Reply:
https://www.notion.so/embraceio/Avoid-code-duplication-in-iOS-Native-layer-for-embrace-io-react-native-otlp-12a7e3c998528037825cd5cfa59805ea